DigitalLearningSolutions.Web/Controllers/TrackingSystem/CentreConfigurationController.cs

```diff
@@ -1,46 +1,80 @@
-        {
-            this.centresDataService = centresDataService;
-            this.customPromptsService = customPromptsService;
-        }
-
-        public IActionResult Index()
-        {
-            var centreId = User.GetCentreId();
-
-            var centreDetails = centresDataService.GetCentreDetailsById(centreId);
-
-            var model = new CentreConfigurationViewModel(centreDetails);
-
-            return View("Index", model);
-        }
-
-        [Route("RegistrationPrompts")]
-        public IActionResult RegistrationPrompts()
-        {
-            var centreId = User.GetCentreId();
-
-            var customPrompts = customPromptsService.GetCustomPromptsForCentreByCentreId(centreId);
-
-            var model = new DisplayRegistrationPromptsViewModel(customPrompts);
-
-            return View(model);
-        }
-    }
-}
+namespace DigitalLearningSolutions.Web.Controllers.TrackingSystem
+{
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.ViewModels.TrackingSystem;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Mvc;
+
+    [Authorize(Policy = CustomPolicies.UserCentreAdminOnly)]
+    [Route("/TrackingSystem/CentreConfiguration")]
+    public class CentreConfigurationController : Controller
+    {
+        private readonly ICentresDataService centresDataService;
+        private readonly ICustomPromptsService customPromptsService;
+
+        public CentreConfigurationController
+        (
+            ICentresDataService centresDataService,
+            ICustomPromptsService customPromptsService
+        )
+        {
+            this.centresDataService = centresDataService;
+            this.customPromptsService = customPromptsService;
+        }
+
+        public IActionResult Index()
+        {
+            var centreId = User.GetCentreId();
+
+            var centreDetails = centresDataService.GetCentreDetailsById(centreId);
+
+            var model = new CentreConfigurationViewModel(centreDetails);
+
+            return View("Index", model);
+        }
+
+        [Route("RegistrationPrompts")]
+        public IActionResult RegistrationPrompts()
+        {
+            var centreId = User.GetCentreId();
+
+            var customPrompts = customPromptsService.GetCustomPromptsForCentreByCentreId(centreId);
+
+            var model = new DisplayRegistrationPromptsViewModel(customPrompts);
+
+            return View(model);
+        }
+
+        [HttpGet]
+        [Route("CentreManagerDetails/Edit")]
+        public IActionResult EditCentreManagerDetails()
+        {
+            var centreId = User.GetCentreId();
+
+            var centreDetails = centresDataService.GetCentreDetailsById(centreId);
+
+            var model = new CentreManagerDetailsViewModel(centreDetails);
+
+            return View("CentreManagerDetails", model);

```

Comment:
Minor, but should the view & viewmodel have "Edit" in the name?

Reply:
I was conflicted on this. 

- On one hand, there is nothing stopping this being about any general centre manager details. We also have only one page (currently) that uses it, so its quite obvious what it links to, despite not matching the route name explicitly, and I doubt we will have any other pages about these details.
- One the other, it is currently not conforming to route name, and other edit pages use the Edit prefix.

Seeing how the route is /CentreManagerDetails/Edit, should the name change to CentreManagerDetailsEditView... or do we want EditCentreManagerDetailsView... ?

Reply:
If it makes sense to change the route to /EditCentreManagerDetails to make the namings a bit nicer/more consistent, I'd go for that. 👍

If we ever need to  use more generic CentreManagerDetails components, we should put them into Shared folders, but this is not very likely to happen in the near future (based on all the future work I've seen so far) so probably YAGNI